File: cses/leetcode/cpp/3147.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    int maximumEnergy(vector<int> &energy, int k)
    {
        int n = energy.size();
        vector<int> dp(n, 0);
        for (int i = 0; i < n; i++)
        {
            int before = 0;
            if (i - k >= 0)
            {
                before = dp[i - k];
            }
            dp[i] = before + energy[i];
        }
        cout << "Out the dp" << endl;
        for (int i = 0; i < n; i++)
        {
            cout << dp[i] << " ";
        }
        cout << endl;

        int max_value = *max_element(energy.end() - k, energy.end());
        return max_value;
    }
};

File: cses/leetcode/javascript/3147.js
/**
 * @param {number[]} energy
 * @param {number} k
 * @return {number}
 */
var maximumEnergy = function(energy, k) {
    let n = energy.length;
    let dp = new Array(n);
    for (let index = 0; index < array.length; index++) {
        const element = array[index];
        let before = 0;
        if(index-k>=0){
            before = dp[index-k];
        }
        dp[index] = max(element,before+element)
    }
    let max_value = 0;
    for (let index = n-1; index >=n-1-k; index--) {
       max_value = max(max_value, dp[index]); 
    }
    return max_value;
};